packages/data-point-express/lib/middleware.js

```diff
@@ -42,20 +42,23 @@ function createErrorMessage(err) {
   };
 }
 
-function resolveReducer(dataPoint, reducer, options, res, initialValue = {}) {
-  dataPoint
-    .transform(reducer, initialValue, options)
-    .then(acc => {
-      if (typeof acc.value === "string") {
-        res.send(acc.value);
-        return;
-      }
-      res.json(acc.value);
-    })
-    .catch(err => {
-      res.status(400).send(createErrorMessage(err));
-    })
-    .done();
+async function resolveReducer(
+  dataPoint,
+  reducer,
+  options,
+  res,
+  initialValue = {}
+) {
+  try {
+    const acc = await dataPoint.transform(reducer, initialValue, options);
+    if (typeof acc.value === "string") {
+      res.send(acc.value);
```

Comment:
I think you can also use here only res.send - without using res.json https://blog.fullstacktraining.com/res-json-vs-res-send-vs-res-end-in-express/

Reply:
yes you can, they are very similar, res.json assumes you are sending JSON and automatically will add the json header. so its just more explicit.

Reply:
Ok my point was to simply do:

`res.send(acc.value)`

without checking for type and using res.json

Reply:
ah ok, good call!!, yes possible, outside of the scope of the PR, but welcome a new PR for sure